app/javascript/packs/index.js

```diff
@@ -1,10 +1,15 @@
 console.log('Hello from index.js')
 
-var locationform = document.getElementById("location");
-
-function addNewLocationForm() {
+function addNewLocationForm(event) {
     console.log('add new form clicked 000')
-    console.log('form is ', locationform)
+    var locationForm = JSON.parse(event.dataset.fieldset);
+    var newId = new Date().getTime();
+    var regexp = new RegExp("new_" + event.dataset.association, "g");
```

Comment:
like I mentioned that nested attributes uses some arbitrary values for it's child index, the default index that was set in the helper is new_{association}, equivalent to `new_location`, this line just replaces that `new_location` index and sets it to another unique but arbitrary value, e.g. the current time. Any unique value would work, as long as it doesn't already exist as a location_id in the database